### PR TITLE
feat(Text): Create new Text component for displaying text content

### LINKED ISSUE #3842: Create Text component
### Description
A new component for displaying text.

Prop for font size = `l,xl, s` etc?

Emphasis?

Perhaps follow this approach https://reactnative.dev/docs/text to improve Dx

### PR DESCRIPTION
**Pull request by markhigham**
## Related issue

#3842

## Overview

A new experimental Text component `TextE` for laying out text content in a consistent way

## Reason

Users of the design system haven't been given guidance about how to layout their text content. This allows for a proliferation of individual approaches. Some based on GDS others on team preferences. This component is a move towards standardsizing how we approach typography

## Work carried out

- [x] New component
- [x] New stories with basic guidance
- [x] Fix regression with `Group` component to prevent overflow 

## Screenshot

See Contentful ...

### CODE REVIEW

File: packages/react-component-library/src/components/Text/Questions.mdx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Meta, Source } from '@storybook/blocks'
+
+import {Text} from './Text'
+
+<Meta title="Questions" />
+
+# Text component questions
+
+- Do we have a defined vertical rhythm?
```
Comment (markhigham) [inline packages/react-component-library/src/components/Text/Questions.mdx]: @m7kvqbe1 ^^

File: packages/react-component-library/src/components/Text/Text.test.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { render } from '@testing-library/react'
+
+import { TextE } from '.'
+import { getTextStyles } from './textStyles'
+
+describe('Text', () => {
+  it('render with default styles', () => {
+    const { container } = render(<TextE>Content</TextE>)
+    expect(container).toHaveTextContent('Content')
+    // expect(container.firstChild).toHaveStyleRule('display', 'flex')
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/Text/Text.test.tsx]: Does this need removing?

File: packages/react-component-library/src/styled-components/GlobalStyle.tsx [hunk 3]
```diff
@@ -94,6 +94,8 @@ const Fonts = createGlobalStyle`
     font-size: ${fontSize('base')};
     line-height: 1.5;
   }
+
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/styled-components/GlobalStyle.tsx]: Redundant newlines

**Review by m7kvqbe1 — APPROVED** [review]
Very nice this looks great!

**Pull request merged**